fearless_simd/src/lib.rs

```diff
@@ -50,6 +50,18 @@
 //! given target features enabled.
 //! As such, most functions which you write when using Fearless SIMD should have the `#[inline(always)]` attribute.
 //!
+//! There is a rule of thumb for how to achieve things in Fearless SIMD:
+//!
+//! - All SIMD functions need `#[inline(always)]`.
+//! - Use [`dispatch!`] when calling SIMD code from non-SIMD code.
+//! - Use [`vectorize()`](Simd::vectorize) when calling SIMD from SIMD if you don't want to force inlining.
```

Comment:
I love tests as concrete examples. A sequence of "function composition" tests with various combinations of non simd calling simd. Simd calling non simd. And simd calling simd would be awesome to have. 

Sorry I'm on my phone hence personal account but I think this is a great improvement and can land as is. 
Thank you!